gnosis/test/Makefile

```diff
@@ -6,7 +6,7 @@ TMPLS:=../module-tmpl.k spec-tmpl.k
 SPEC_GROUP:=gnosis-test
 SPEC_INI:=api-test.ini
 
-KPROVE_OPTS:=--smt-prelude $(abspath $(dir $(abspath $(lastword $(MAKEFILE_LIST))))/../evm.smt2)
+KPROVE_OPTS:=--smt-prelude $(abspath $(dir $(abspath $(lastword $(MAKEFILE_LIST))))/../evm.smt2) --deterministic-functions
```

Comment:
why `--deterministic-functions` option only for this?   shouldn't we put it for all others?

Reply:
fixed. For all gnosis specs. For others I'll open a separate PR just in case.